src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -162,6 +177,20 @@ public async Task<T> OpenContextForWriteAsync<T>(Func<IWorkspaceProjectContextAc
                 }
             }
 
+            private void ApplyBuildSnapshotChanged(IProjectVersionedValue<(ConfiguredProject project, IProjectBuildSnapshot buildSnapshot)> update)
+            {
+                List<string> commandLineArguments = new List<string>();
+
+                update.Value.buildSnapshot.TargetOutputs.TryGetValue("CompileDesignTime", out var keyValuePairsCommandLineOptions);
```

Comment:
Please make this way more efficient; as spoken about earlier this is very sensitive to allocations and LOH allocations, the List above will be resized many times over. I think we want a more efficient way to go from the IEnumerable<string> -> ImmutableArray.

Reply:
done